src/web_buffer.c

```diff
@@ -187,43 +187,52 @@ void buffer_rrd_value(BUFFER *wb, calculated_number value)
 // generate a javascript date, the fastest possible way...
 void buffer_jsdate(BUFFER *wb, int year, int month, int day, int hours, int minutes, int seconds)
 {
-	//         10        20        30      = 35
+  //         10        20        30      = 35
 	// 01234567890123456789012345678901234
 	// Date(2014,04,01,03,28,20)
 
 	buffer_need_bytes(wb, 30);
 
-	char *b = &wb->buffer[wb->len];
-
-	int i = 0;
-	b[i++]='D';
-	b[i++]='a';
-	b[i++]='t';
-	b[i++]='e';
-	b[i++]='(';
-	b[i++]= (char) (48 + year / 1000); year -= (year / 1000) * 1000;
-	b[i++]= (char) (48 + year / 100); year -= (year / 100) * 100;
-	b[i++]= (char) (48 + year / 10);
-	b[i++]= (char) (48 + year % 10);
-	b[i++]=',';
-	b[i]= (char) (48 + month / 10); if(b[i] != '0') i++;
-	b[i++]= (char) (48 + month % 10);
-	b[i++]=',';
-	b[i]= (char) (48 + day / 10); if(b[i] != '0') i++;
-	b[i++]= (char) (48 + day % 10);
-	b[i++]=',';
-	b[i]= (char) (48 + hours / 10); if(b[i] != '0') i++;
-	b[i++]= (char) (48 + hours % 10);
-	b[i++]=',';
-	b[i]= (char) (48 + minutes / 10); if(b[i] != '0') i++;
-	b[i++]= (char) (48 + minutes % 10);
-	b[i++]=',';
-	b[i]= (char) (48 + seconds / 10); if(b[i] != '0') i++;
-	b[i++]= (char) (48 + seconds % 10);
-	b[i++]=')';
-	b[i]='\0';
-
-	wb->len += i;
+	char *b = &wb->buffer[wb->len], *p;
+  unsigned int *q = (unsigned int *)b;  
+
+  #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    *q++ = 0x65746144;  // "Date" backwards.
+  #else
+    *q++ = 0x44617465;  // "Date"
+  #endif
+  p = (char *)q;
+
+  *p++ = '(';
+  *p++ = '0' + year / 1000; year %= 1000;
+  *p++ = '0' + year / 100;  year %= 100;
+  *p++ = '0' + year / 10;
+  *p++ = '0' + year % 100;
```

Comment:
This line is wrong.
It should be:

```c
*p++ = '0' + year % 10;
```

To reproduce the problem you have to access the old dashboard: `http://your.netdata:19999/old/`

I am submitting a patch now.